### PR TITLE
fix: Fold the end line of template / block if it's braceless

### PR DESCRIPTION
**Pull request by tanishiking**
fix: https://github.com/scalameta/metals/issues/4190

FoldRangeProider for Scala3 with braceless syntax had a problem:

If clientCapabilities set `lineFoldingOnly: true`

```ts
	/**
	 * If set, the client signals that it only supports folding complete lines.
	 * If set, client will ignore specified `startCharacter` and `endCharacter`
	 * properties in a FoldingRange.
	 */
	lineFoldingOnly?: boolean;
```

Metals minus 1 from the end line of foldRange response
https://github.com/scalameta/metals/blob/81b014d4deb62f55fdef99f266e2aedf02efcc90/metals/src/main/scala/scala/meta/internal/parsing/FoldingRangeProvider.scala#L86-L91
so that Metals can preserve the last line containing `}`.

However, in Scala3, braceless syntax doesn't have `}` in the last line for no reason.
As a result, for example, if we fold the following code,

```scala
def foo =
  ???
  ???
  ???
```

we will see

```scala
def foo = ...
  ???
```

(screen-shot from the issue)
螢幕截圖 2022-07-26 上午11 36 19
螢幕截圖 2022-07-26 上午11 36 27

---

This commit fixes this behavior by not adjusting the last line if it's braceless syntax.

### CODE REVIEW

File: metals/src/main/scala/scala/meta/internal/parsing/FoldingRangeExtractor.scala [hunk 1]
```diff
@@ -148,21 +148,21 @@ final class FoldingRangeExtractor(
   }
 
   private object Foldable {
-    def unapply(tree: Tree): Option[Position] =
+    def unapply(tree: Tree): Option[(Position, Boolean)] =
```
Comment (tanishiking) [inline metals/src/main/scala/scala/meta/internal/parsing/FoldingRangeExtractor.scala:151]: Returns a flag whether FoldingProvider should adjust the last line or not

File: tests/unit/src/test/resources/foldingRange-scala3-foldLineOnly/input/Givens.scala [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      "o" +
+      "o"
+
+given Foo[String] with 
+  def a: String = 
+    "f" +
+      "o" +
+      "o"
+
+given x: AnyRef with
```
Comment (tgodzik) [inline tests/unit/src/test/resources/foldingRange-scala3-foldLineOnly/input/Givens.scala:16]: Could you add one test with end marker?

**Review by tgodzik — CHANGES_REQUESTED** [review]

File: tests/unit/src/test/resources/foldingRange-scala3-foldLineOnly/input/Givens.scala (hunk 2 above)
Comment (tanishiking) [reply tests/unit/src/test/resources/foldingRange-scala3-foldLineOnly/input/Givens.scala:16]: Added https://github.com/scalameta/metals/pull/4191/commits/81e7f54d7aa5705a66caba4a677dc6f64a41b103 :+1:

**Review by tgodzik — APPROVED** [review]
LGTM!

**Pull request merged**